### ISSUE TITLE
Estos son los themas

### ISSUE THREAD

**Issue by appsdevelopmentvlneng-create** [JuanCuevasLN/E-commerce#6]
Theme
popularfx-child

style.css
wp-content/themes/popularfx-child/style.css

Parent Theme
popularfx

style.css
wp-content/themes/popularfx/style.css

Template File
popularfx/page.php

Template Hierarchy
page-carrito.php
page-8.php
page.php
singular.php
index.php
Template Parts
popularfx-child/header.php
popularfx/template-parts/content-page.php
popularfx/sidebar.php
popularfx/footer.php
Body Classes
admin-bar
logged-in
no-customize-support
page
page-id-8
page-template-default
popularfx-body
theme-popularfx
woocommerce-active
woocommerce-cart
woocommerce-no-js
woocommerce-page
wp-child-theme-popularfx-child
wp-singular
wp-theme-popularfx

**Comment by JuanCuevasLN** [comment]
<?php
/* Template Name: Carrito personalizado */
defined( 'ABSPATH' ) || exit;

get_header(); ?>

<main id="primary" class="site-main custom-cart">
  <h1 class="page-title">Mi carrito</h1>

  <?php
  // Renderiza el carrito oficial de WooCommerce
  echo do_shortcode('[woocommerce_cart]');
  ?>
</main>

<?php get_footer(); ?>

**Comment by JuanCuevasLN** [comment]
<aside id="cart-aside" class="cart-aside hidden">
  <div class="cart-overlay"></div>
  <div class="cart-panel">
    <header class="cart-header">
      <h2>Tu carrito</h2>
      <button id="close-cart" aria-label="Cerrar carrito">&times;</button>
    </header>

    <div id="cart-items" class="cart-items">
      
    </div>

    <footer class="cart-footer">
      <p class="cart-total">Total: <span id="cart-total-amount">0</span></p>
      <a href="<?php echo wc_get_cart_url(); ?>" class="btn-checkout">Ver carrito</a>
    </footer>
  </div>
</aside>

**Comment by JuanCuevasLN** [comment]
.cart-aside {
  position: fixed;
  inset: 0;
  display: flex;
  justify-content: flex-end;
  z-index: 9999;
  transition: opacity 0.3s ease;
}

.cart-aside.hidden {
  opacity: 0;
  pointer-events: none;
}

.cart-overlay {
  flex: 1;
  background: rgba(0,0,0,0.5);
}

.cart-panel {
  width: 400px;
  max-width: 90%;
  background: #fff;
  display: flex;
  flex-direction: column;
  height: 100%;
  transform: translateX(100%);
  transition: transform 0.3s ease;
}

.cart-aside.active .cart-panel {
  transform: translateX(0);
}

.cart-header, .cart-footer {
  padding: 1rem;
  border-bottom: 1px solid #eee;
}

.cart-items {
  flex: 1;
  overflow-y: auto;
  padding: 1rem;
}

**Comment by JuanCuevasLN** [comment]
function enqueue_cart_aside_scripts() {
  wp_enqueue_script('cart-aside', get_stylesheet_directory_uri() . '/js/cart-aside.js', ['jquery'], null, true);
  wp_localize_script('cart-aside', 'cartAside', [
    'ajax_url' => admin_url('admin-ajax.php'),
  ]);
}
add_action('wp_enqueue_scripts', 'enqueue_cart_aside_scripts');

**Comment by JuanCuevasLN** [comment]
jQuery(document).ready(function($) {
  const $cartAside = $('#cart-aside');
  const $cartItems = $('#cart-items');
  const $cartTotal = $('#cart-total-amount');

  function loadCart() {
    $.ajax({
      url: cartAside.ajax_url,
      method: 'POST',
      data: { action: 'get_cart_contents' },
      success: function(response) {
        if (response.success) {
          $cartItems.html(response.data.html);
          $cartTotal.text(response.data.total);
        }
      }
    });
  }

  // Abrir carrito (ejemplo: botón con clase .open-cart)
  $(document).on('click', '.open-cart', function() {
    $cartAside.addClass('active').removeClass('hidden');
    loadCart();
  });

  // Cerrar carrito
  $('#close-cart, .cart-overlay').on('click', function() {
    $cartAside.removeClass('active').addClass('hidden');
  });
});

**Comment by JuanCuevasLN** [comment]
add_action('wp_ajax_get_cart_contents', 'get_cart_contents');
add_action('wp_ajax_nopriv_get_cart_contents', 'get_cart_contents');

function get_cart_contents() {
  $items = WC()->cart->get_cart();
  ob_start();
  
  if (!empty($items)) {
    foreach ($items as $cart_item_key => $cart_item) {
      $product = $cart_item['data'];
      ?>
      <div class="cart-item">
        <img src="<?php echo get_the_post_thumbnail_url($product->get_id(), 'thumbnail'); ?>" alt="">
        <div class="item-info">
          <p><?php echo $product->get_name(); ?></p>
          <p>Cantidad: <?php echo $cart_item['quantity']; ?></p>
          <p>Precio: <?php echo wc_price($product->get_price()); ?></p>
        </div>
      </div>
      <?php
    }
  } else {
    echo '<p>Tu carrito está vacío</p>';
  }

  $html = ob_get_clean();

  wp_send_json_success([
    'html' => $html,
    'total' => WC()->cart->get_total(),
  ]);
}
